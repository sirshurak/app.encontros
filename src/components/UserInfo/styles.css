#user-to-chat.horizontal {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

#user-to-chat.horizontal figure {
    margin: 0;
    margin-bottom: 5px;
    width: 80px;
    height: 80px;
}

#user-to-chat img {
    width: 100%;
    height: 100%;
    border-radius: 50%;
}

#user-to-chat.horizontal .more-info {
    display: none;
}

#user-to-chat h3 {
    font-size: 15px;
}

#user-to-chat.vertical {
    display: grid;
    grid-template-columns:80px 1fr;
    grid-template-rows: 1fr;
    justify-content: center;
    align-items: flex-start;
    border-style: solid;
    border-width: 1px;
    border-radius: 15px;
    background-color: white;
}

#user-to-chat.vertical figure {
    margin: 0;
    align-self: center;
}

#user-to-chat.vertical figure img {
    height: 60px;
    width: 60px;
}

#user-to-chat.vertical .more-info {
    text-align: left;
    padding: 10px;
}

#user-to-chat.vertical .more-info h3 {
    display: inline-block;
    margin: 0;
}

#user-to-chat.vertical .more-info p {
    margin: 0;
    line-height: 1.5rem;
}