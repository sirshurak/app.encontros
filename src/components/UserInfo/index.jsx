import React from "react";
import PropTypes from "prop-types";
import "./styles.css";

function UserInfo({
  variant,
  photo,
  username,
  age,
  ageTitle,
  gender,
  sexualOrientation,
  city,
}) {
  return (
    <section id="user-to-chat" className={variant}>
      <figure>
        <img src={photo} alt="" />
      </figure>
      <div className="info">
        {variant === "horizontal" && <h3 className="username">{username}</h3>}
        <div className="more-info">
          {variant === "vertical" && <h3 className="username">{username}</h3>}
          <span className="age">
            {", "} {age} {ageTitle}
          </span>
          <p>
            <span className="gender">{gender}</span>
            <span className="sexual-orientation">
              {", "}
              {sexualOrientation}
            </span>
          </p>
          <p>
            <span className="city">{city}</span>
          </p>
        </div>
      </div>
    </section>
  );
}

UserInfo.propTypes = {
  variant: PropTypes.oneOf(["horizontal", "vertical"]).isRequired,
  photo: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  ageTitle: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  sexualOrientation: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
};

export default UserInfo;
