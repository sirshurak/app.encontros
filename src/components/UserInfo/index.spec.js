import { shallow, mount } from 'enzyme';
import UserInfo from './index';

const user = {
    photo: 'https://picsum.photos/200/?image=26',
    username: 'username',
    city: 'Test City',
    gender: 'Man',
    sexualOrientation: 'Heterosexual',
    age: 27
}

describe('UserInfo component', () => {
    it('should match shallow vertical snapshop', () => {
        const component = shallow(<UserInfo variant="vertical" ageTitle="anos" {...user}/>)
        expect(component).toMatchSnapshot();
    });

    it('should match shallow horizontal snapshop', () => {
        const component = shallow(<UserInfo variant="horizontal" ageTitle="anos" {...user}/>)
        expect(component).toMatchSnapshot();
    });
});