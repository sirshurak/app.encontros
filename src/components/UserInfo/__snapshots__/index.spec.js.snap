// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component should match shallow horizontal snapshop 1`] = `
<section
  className="horizontal"
  id="user-to-chat"
>
  <figure>
    <img
      alt=""
      src="https://picsum.photos/200/?image=26"
    />
  </figure>
  <div
    className="info"
  >
    <h3
      className="username"
    >
      username
    </h3>
    <div
      className="more-info"
    >
      <span
        className="age"
      >
        , 
         
        27
         
        anos
      </span>
      <p>
        <span
          className="gender"
        >
          Man
        </span>
        <span
          className="sexual-orientation"
        >
          , 
          Heterosexual
        </span>
      </p>
      <p>
        <span
          className="city"
        >
          Test City
        </span>
      </p>
    </div>
  </div>
</section>
`;

exports[`UserInfo component should match shallow vertical snapshop 1`] = `
<section
  className="vertical"
  id="user-to-chat"
>
  <figure>
    <img
      alt=""
      src="https://picsum.photos/200/?image=26"
    />
  </figure>
  <div
    className="info"
  >
    <div
      className="more-info"
    >
      <h3
        className="username"
      >
        username
      </h3>
      <span
        className="age"
      >
        , 
         
        27
         
        anos
      </span>
      <p>
        <span
          className="gender"
        >
          Man
        </span>
        <span
          className="sexual-orientation"
        >
          , 
          Heterosexual
        </span>
      </p>
      <p>
        <span
          className="city"
        >
          Test City
        </span>
      </p>
    </div>
  </div>
</section>
`;
